sds_data_manager/lambda_code/SDSCode/upload_api.py

```diff
@@ -16,29 +17,44 @@
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
-s3 = boto3.client("s3")
+BUCKET_NAME = os.environ["S3_BUCKET"]
+S3_CLIENT = boto3.client("s3")
 
 
-def _generate_signed_upload_url(key_path, tags=None):
+def _generate_signed_upload_response(key_path, tags=None):
     """Create a presigned url for a file in the SDS storage bucket.
 
-    :param key_path: Required.  A string representing the name of the object to upload.
-    :param tags: Optional.  A dictionary that will be stored in the S3 object metadata.
+    Parameters
+    ----------
+    key_path : str
+        The fully qualified path of the object to upload.
+    tags : dict, optional
+         Additional S3 object metadata to add to the object.
 
-    :return: A URL string if the file was found, otherwise None.
+    Returns
+    -------
+    Response with status code and a pre-signed URL for the object.
     """
-    bucket_name = os.environ["S3_BUCKET"]
-    url = boto3.client("s3").generate_presigned_url(
+    url = S3_CLIENT.generate_presigned_url(
         ClientMethod="put_object",
         Params={
-            "Bucket": bucket_name,
+            "Bucket": BUCKET_NAME,
             "Key": key_path,
             "Metadata": tags or dict(),
         },
         ExpiresIn=3600,
     )
 
-    return url
+    if url is None:
+        return {
+            "statusCode": 400,
+            "body": json.dumps(
+                "A pre-signed URL could not be generated. Please ensure that the "
+                "file name matches mission file naming conventions."
```

Comment:
Please feel free to reword this
```suggestion
                "A pre-signed URL could not be generated. Please ensure that the "
                f"file name matches mission file naming convention, {FILENAME_CONVENTION} or SPICE data file name has these extensions - {some_list_var}"
```

Reply:
👍 I added the filename convention and that it should be a valid SPICE file.